src/ucs/datastruct/callbackq.c

```diff
@@ -455,6 +455,7 @@ void ucs_callbackq_cleanup(ucs_callbackq_t *cbq)
 {
     ucs_callbackq_priv_t *priv = ucs_callbackq_priv(cbq);
 
+    ucs_callbackq_purge_fast(cbq);
```

Comment:
why needed? seems like duplication of L460

Reply:
in ucp_worker_destroy(), ucp_worker_close_ifaces() would not (indirectly) call ucs_callbackq_remove() (which calls ucs_callbackq_purge_fast())to disable iface progress because iface progress might have been disabled earlier w/ this PR. so we may finally encounter [this assertion failure](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=34885&view=logs&j=74876da5-c0e8-5509-4929-d550f147815a&t=ad691f47-6e94-5716-c2cf-10e47f6207fd&l=21485) in stage of uct_worker_destroy() (which invokes ucs_callbackq_cleanup()) w/o this statement.

Reply:
IMO `ucs_callbackq_purge_fast()` should be called *before* not *after* `ucs_callbackq_disable_proxy()` to make sure all marked-for-removal elements have been cleared beforehand in order to suppress [this assertion failure](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=34885&view=logs&j=74876da5-c0e8-5509-4929-d550f147815a&t=ad691f47-6e94-5716-c2cf-10e47f6207fd&l=21485). w/ addition of L458, we can delete L460 I think.

Reply:
> IMO `ucs_callbackq_purge_fast()` should be called _before_ not _after_ `ucs_callbackq_disable_proxy()` to make sure all marked-for-removal elements have been cleared beforehand in order to suppress [this assertion failure](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=34885&view=logs&j=74876da5-c0e8-5509-4929-d550f147815a&t=ad691f47-6e94-5716-c2cf-10e47f6207fd&l=21485). w/ addition of L458, we can delete L460 I think.

Agree; I think that we should also add assert to ucs_callbackq_disable_proxy() that when it's called, fast_remove_mask==0.

Reply:
done